plugins/editorconfig/src/org/editorconfig/language/services/impl/EditorConfigIgnoreCaseMap.kt

```diff
@@ -0,0 +1,27 @@
+// Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package org.editorconfig.language.services.impl
+
+import java.util.*
+import java.util.function.BiFunction
+import java.util.function.Function
+
+class EditorConfigIgnoreCaseMap<T> : LinkedHashMap<String, T>() {
+  override operator fun get(key: String) = super.get(key.toLowerCase(Locale.ROOT))
```

Comment:
A string allocation on each operation with a map which is the core data structure for getting information about elements in the file doesn't really sound good.
I'd suggest to use a hash map from trove library with a custom hashing strategy with a com.intellij.util.text.CaseInsensitiveStringHashingStrategy